ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot

```diff
@@ -35,28 +36,51 @@ Prepare Codeflare-sdk E2E Test Suite
     END
     Enable Component    codeflare
     Enable Component    ray
-    Create Directory    %{WORKSPACE}/codeflare-sdk-e2e-logs
+
+     ${result} =    Run Process  virtualenv -p python3.9 ${VIRTUAL_ENV_NAME}
+    ...    shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
+    IF    ${result.rc} != 0
+        FAIL    Unable to setup Python virtual environment
+    END
+
     RHOSi Setup
 
 Teardown Codeflare-sdk E2E Test Suite
     [Documentation]    Teardown codeflare-sdk E2E Test Suite
     Disable Component    codeflare
     Disable Component    ray
+
+    ${result} =    Run Process  rm -rf ${VIRTUAL_ENV_NAME}
+    ...    shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
+    IF    ${result.rc} != 0
+        FAIL   Unable to cleanup Python virtual environment
+    END
+
+    ${result} =    Run Process    poetry env use 3.11
+    ...    shell=true    stderr=STDOUT
+    IF    ${result.rc} != 0
+        FAIL   Unable to switch python environment 3.11
+    END
+
+    ${result} =    Run Process  rm -rf ${CODEFLARE-SDK_DIR}
+    ...    shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
+    IF    ${result.rc} != 0
+        FAIL   Unable to cleanup directory ${CODEFLARE-SDK_DIR}
+    END
+    
     RHOSi Teardown
 
 Run Codeflare-sdk E2E Test
     [Documentation]    Run codeflare-sdk E2E Test
     [Arguments]    ${test_name}
     Log To Console    Running codeflare-sdk test: ${test_name}
-    ${result} =    Run Process    go test -timeout 30m -v ./tests/e2e -run ${test_name}
+    ${result} =    Run Process  source ${VIRTUAL_ENV_NAME}/bin/activate && cd codeflare-sdk && poetry env use 3.9 && poetry install --with test,docs && poetry run pytest -v -s ./tests/e2e/${TEST_NAME} --timeout\=600 && deactivate
```

Comment:
Maybe can we somehow parameterize versions of Python? For example as test suite variable?

Reply:
It would be good to make it on main branch.
Here for 2.8.x we can use hardcoded version as it won't change in patch releases.

Reply:
Hey @jiripetrlik I have added a test-suite variable named `PYTHON_VERSION` to parse python_version in test, Please review!
Thank you @sutaakar @jiripetrlik for reviewing this PR!